models/amazon_ads__ad_report.sql

```diff
@@ -12,15 +12,15 @@ account_info as (
 ),
 
 portfolios as (
-    select
-    {% if var('amazon_ads__portfolio_history_enabled', True) %}
-        *
-        from {{ var('portfolio_history') }}
-        where is_most_recent_record = True
-    {% else %}
-        null as portfolio_name,
-        null as portfolio_id
-    {% endif %}
+    select 
+    {% if var('amazon_ads__portfolio_history_enabled', True) %} 
+        * 
+        from {{ var('portfolio_history') }} 
+        where is_most_recent_record = True 
+    {% else %} 
+        null as portfolio_name, 
+        null as portfolio_id 
+    {% endif %} 
```

Comment:
I really like this piece of code. This ensures we only need to conditional in this one location. In other implementations we have had to include the conditional both here **and** within the join **and** in the join. This is way more elegant. Great job!

I was thinking though, this snippet is reused in every end model. We could probably get around doing this just once in an intermediate model and then have the end models just reference that intermediate model. This way if there is ever a change needed to this piece of code, then it just needs to be done in one intermediate model as opposed to all the end models.

Can you break this out into an intermediate model and just reference that model in the end models.

Reply:
Updated!